tensorboard/webapp/metrics/store/metrics_types.ts

```diff
@@ -126,10 +126,18 @@ export type CardMetadataMap = Record<
   CardMetadata
 >;
 
+export enum CardSelectionState {
+  GLOBAL,
+  ENABLED,
+  DISABLED,
+}
+
 export type CardState = {
   dataMinMax: MinMaxStep;
   userMinMax: MinMaxStep;
   timeSelection: TimeSelection;
+  stepSelection: CardSelectionState;
+  rangeSelection: CardSelectionState;
```

Comment:
While these could technically become
```suggestion
  stepSelectionEnabled: boolean | undefined;
  rangeSelectionEnabled: boolean | undefined;
```

I don't like the idea of introducing a third state to boolean values.